pkg/landscaper/controllers/execution/controller.go

```diff
@@ -65,7 +65,28 @@ func (c *controller) Reconcile(ctx context.Context, req reconcile.Request) (reco
 	}
 
 	if lsv1alpha1helper.IsCompletedExecutionPhase(exec.Status.Phase) {
-		logger.V(7).Info("execution item is already in completion phase")
+		op := execution.NewOperation(operation.NewOperation(logger, c.client, c.scheme, c.eventRecorder), exec, false)
+		phaseChanged, err := handleDeployItemPhaseChanges(ctx, op, logger, exec)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+		if phaseChanged {
+			if exec.Status.Phase == lsv1alpha1.ExecutionPhaseSucceeded {
+				// phase changed to Succeeded, it might be necessary to generate the exports again
+				logger.V(5).Info("phase changed to '%s', compute deploy item exports again", string(lsv1alpha1.ExecutionPhaseSucceeded))
+				deployItems, err := op.ListManagedDeployItems(ctx)
```

Comment:
we list the deployitems 2 times.
once in the `handleDeployItemPhaseChanges` function and then again here when the status has changed to succeeded.

How about doing all the code in the operation like the `operation.Reconcile` then we have everything in one function and one package.
and could reuse the already listed deployitems